internal/client.go

```diff
@@ -626,6 +630,528 @@ type (
+		// policy will be used.
+		Trigger(ctx context.Context, overlap enumspb.ScheduleOverlapPolicy) error
+
+		// Pause the Schedule will also overwrite the Schedules current note with the new note.
+		// If an empty note is passed the Schedules note will be set to 'Paused via Go SDK'.
+		Pause(ctx context.Context, note string) error
+
+		// Unpause the Schedule will also overwrite the Schedules current note with the new note.
+		// If an empty note is passed the Schedules note will be set to 'Unpaused via Go SDK'.
+		Unpause(ctx context.Context, note string) error
+	}
+
+	// ScheduleActionResult describes when a schedule action took place
+	ScheduleActionResult struct {
+		// ScheduleTime - Time that the Action was scheduled for, including jitter.
+		ScheduleTime time.Time
+
+		// ActualTime - Time that the Action was actually taken.
+		ActualTime time.Time
+
+		// StartWorkflowResult - If action was ScheduleWorkflowAction, returns the
+		// ID of the workflow.
+		StartWorkflowResult ScheduleWorkflowExecution
+	}
+
+	// ScheduleListEntry
+	ScheduleListEntry struct {
+		// ScheduleID - The business identifier of the schedule.
+		ScheduleID string
+
+		// Spec - Describes when Actions should be taken.
+		Spec *ScheduleSpec
+
+		// Note - Informative human-readable message with contextual notes, e.g. the reason
+		// a Schedule is paused. The system may overwrite this message on certain
+		// conditions, e.g. when pause-on-failure happens.
+		Note string
+
+		// Paused - True if the schedule is paused.
+		Paused bool
+
+		// WorkflowType - If the schedule action is a Wokrflow then
+		// describes what workflow is run.
+		WorkflowType WorkflowType
+
+		// RecentActions- Most recent 10 Actions started (including manual triggers).
+		//
+		// Sorted from older start time to newer.
+		RecentActions []ScheduleActionResult
+
+		// NextActionTimes - Next 10 scheduled Action times.
+		NextActionTimes []time.Time
+
+		// Memo - Non-indexed user supplied information.
+		Memo *commonpb.Memo
+
+		// SearchAttributes - Indexed info that can be used in query of List schedules APIs (only
+		// supported when Temporal server is using advanced visibility). The key and value type must be registered on Temporal server side.
+		// Use GetSearchAttributes API to get valid key and corresponding value type.
+		SearchAttributes *commonpb.SearchAttributes
+	}
+
+	// ScheduleListOptions are the parameters for configuring listing schedules
+	ScheduleListOptions struct {
+		// PageSize - How many results to fetch from the Server at a time.
+		// Optional: defaulted to 1000
+		PageSize int32
+	}
+
+	// ScheduleListIterator represents the interface for
+	// schedule iterator
+	ScheduleListIterator interface {
+		// HasNext return whether this iterator has next value
+		HasNext() bool
+
+		// Next returns the next schedule and error
+		Next() (*ScheduleListEntry, error)
+	}
+
+	// Client for creating Schedules and creating Schedule handles
+	ScheduleClient interface {
```

Comment:
Maybe just `Create`, `List`, and `GetHandle` since you're already scoped to a schedule client

Reply:
it's only scoped to the interface, not the implementation. The implementation could still have collisions if it implemented multiple interfaces.

In this case probably not an issue, but that was my thought process.

Reply:
> The implementation could still have collisions if it implemented multiple interfaces.

Would prefer to optimize for callers than implementers here. `myClient.ScheduleClient().Create()` seems more reasonable than `myClient.ScheduleClient().CreateSchedule()`.